Add DNSSEC checks

jsf9k opened the pull request.

echudow committed:
Added DNSSEC checks

jsf9k committed:
Merge remote-tracking branch 'cisagov/develop' into develop

jsf9k committed:
Make flake8 happy

jsf9k committed:
Make flake8 happy

jsf9k commented:
@echudow, please give me permission to write to the `develop` branch on echudow/trustymail and I'll merge in the latest from upstream and make the CI tests pass.

@IanLee1521 may also need write permission too.  He is a frequent contributer to cisagov/pshtt, cisagov/trustymail, and 18F/domain-scan.

echudow committed:
Leaving None as None for unknowns

echudow committed:
Merge branch 'develop' of https://github.com/cisagov/trustymail into develop

echudow commented:
@jsf9k, I added you and @IanLee1521. Thanks for the help.

jsf9k committed:
Merge branch 'develop' of github.com:echudow/trustymail into develop

jsf9k commented:
@echudow, I'm still not allowed to push to `develop`:
```
$ git push
Enumerating objects: 28, done.
Counting objects: 100% (28/28), done.
Delta compression using up to 8 threads
Compressing objects: 100% (16/16), done.
Writing objects: 100% (16/16), 4.36 KiB | 1.45 MiB/s, done.
Total 16 (delta 8), reused 0 (delta 0)
remote: Resolving deltas: 100% (8/8), completed with 3 local objects.
To github.com:echudow/trustymail.git
 ! [remote rejected] develop -> develop (permission denied)
error: failed to push some refs to 'git@github.com:echudow/trustymail.git'
```

echudow commented:
I added you as a collaborator on [echudow/trustymail](https://github.com/echudow/trustymail) and it says that it invited you, but it says that it's waiting for you to accept the invite. 

I also invited you to a private repo for building custom trust stores for use with pshtt and sslyze if you are ever interested in taking a look.  It's focused on DoD, but the extra intermediate certs could be somewhat useful.

jsf9k commented:
Thanks, @echudow.  The email was in my person email account and somehow I missed it.  I was able to push just now.  I'll also take a look at the other repo.

jsf9k committed:
Bump version from 0.6.9 to 0.7.0

felddy commented on trustymail/domain.py:
```diff
@@ -51,7 +51,7 @@ def format_list(record_list):
     # lists.  This makes a "null" appear in the JSON output for
     # empty lists, as expected.
-    if not record_list:
+    if not record_list or len(record_list) == 0:
```
This code change does not modify the logic.  It will not be executed as a list of length `0` will already evaluate to `false` and short circuit the `or`.

h-m-f-t replied:
Best meta joke about DNSSEC I've seen today

:wave: :eyes: @echudow

felddy replied:
🤣

jsf9k replied:
Reverted in 135833d.

echudow replied:
Sorry, I'm too used to other languages where that check would have been useful.  
@h-m-f-t, I'm sure you'll find some other jokes about DNSSEC to throw my way too.

jsf9k commented:
Based on the results from a full run with the new code, I think this PR is ready to be merged. I can't approve the PR myself, though, since I created it. @echudow or @IanLee1521, can either or you approve this PR?

jsf9k committed:
Remove the len(record_list) == 0 condition

Only lists are passed to the function, and an empty list already
evaluates a False, so the extra check adds nothing.

jsf9k committed:
Add descriptions for added output fields

jsf9k committed:
Return None instead of "Unknown"

We prefer a value of None to represent that a test produced
inconclusive results or was not run at all.

jsf9k committed:
Get rid of a lot of unnecessary () and introduce a space after if

jsf9k committed:
Add LGTM build badges
